src/OnlineRetailPortal.Web/Validations/AddProductRequestValidator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace OnlineRetailPortal.Web.Validations
+{
+    public class AddProductRequestValidator : AbstractValidator<AddProductRequest>
+    {
+        public AddProductRequestValidator()
+        {
+            RuleFor(x => x.UserId)
+            .NotEmpty()
+            .WithMessage("The User ID  cannot be blank.");
+
+            RuleFor(x => x.Id)
+            .NotEmpty()
+            .WithMessage("The Product ID  cannot be blank."); ;
+
+            RuleFor(x => x.Name)
+            .NotEmpty()
+            .Length(2, 20)
+            .WithMessage("The Title cannot be less than 2 characters and more than  20 characters.");
+
+            RuleFor(x => x.Category)
+            .NotEmpty()
+            .WithMessage("The Categorye cannot be blank.");
+
+            RuleFor(x => x.Price.Amount)
+            .NotEmpty()
+            .InclusiveBetween(1, 10000000)
```

Comment:
What if this is the property for sale?
We should not restrict outer bound 
Or we need to expose it as a part of contract that products beyond 10000000
Let’s discuss this tomorrow

Reply:
ok

Reply:
done